packages/admin-next/dashboard/src/routes/draft-orders/draft-order-create/components/create-draft-order-form/create-draft-order-form.tsx

```diff
@@ -0,0 +1,454 @@
+import { getDbAmount } from "../../../../../lib/money-amount-helpers"
+
+import { PricedVariant } from "@medusajs/medusa/dist/types/pricing"
+import { CreateDraftOrderSchema, View } from "./constants"
+import { CreateDraftOrderContext } from "./context"
+import { CreateDraftOrderDetails } from "./create-draft-order-details"
+import { CreateDraftOrderDrawer } from "./create-draft-order-drawer"
+import { CreateDraftOrderSummary } from "./create-draft-order-summary"
+import { CustomItem, ExistingItem } from "./types"
+
+enum Tab {
+  DETAILS = "details",
+  SUMMARY = "summary",
+}
+
+export const CreateDraftOrderForm = () => {
+  const [open, setOpen] = useState(false)
+
+  const [view, setView] = useState<View | null>(null)
+  const [tab, setTab] = useState<Tab>(Tab.DETAILS)
+  const [detailsValidated, setDetailsValidated] = useState(false)
+
+  const [region, setRegion] = useState<Region | null>(null)
+  const [customer, setCustomer] = useState<Customer | null>(null)
+
+  const [sameAsShipping, setSameAsShipping] = useState(true)
+
+  const { t } = useTranslation()
+  const [, setSearchParams] = useSearchParams()
+  const { handleSuccess } = useRouteModal()
+
+  const form = useForm<z.infer<typeof CreateDraftOrderSchema>>({
+    defaultValues: {
+      email: "",
+      region_id: "",
+      shipping_method: {
+        amount: "",
+        custom_amount: "",
+        option_id: "",
+        option_title: "",
+      },
+      shipping_address: {
+        address_1: "",
+        address_2: "",
+        city: "",
+        company: "",
+        country_code: "",
+        first_name: "",
+        last_name: "",
+        phone: "",
+        postal_code: "",
+        province: "",
+      },
+      billing_address: null,
+      existing_items: [],
+      custom_items: [],
+      customer_id: "",
+      notification_order: true,
+    },
+    resolver: zodResolver(CreateDraftOrderSchema),
+  })
+
+  const {
+    clearErrors,
+    formState: { isDirty },
+  } = form
+
+  const {
+    append: createCustomItem,
+    remove: deleteCustomItem,
+    fields: customItems,
+  } = useFieldArray({
+    control: form.control,
+    name: "custom_items",
+    keyName: "ci_id",
+  })
+
+  const {
+    append: createExistingItem,
+    remove: deleteExistingItem,
+    update: updateExistingItem,
+    fields: existingItems,
+  } = useFieldArray({
+    control: form.control,
+    name: "existing_items",
+    keyName: "ei_id",
+  })
+
+  const { mutateAsync, isLoading } = useAdminCreateDraftOrder()
+
+  const handleSubmit = form.handleSubmit(async (values) => {
+    let {
+      shipping_address,
+      billing_address,
+      existing_items,
+      custom_items,
+      shipping_method,
+      email,
+      notification_order,
+      ...rest
+    } = values
+
+    const emailValue = email || customer?.email
+
+    if (!emailValue) {
+      form.setError("email", {
+        type: "manual",
+        message: "Email is required",
+      })
```

Comment:
this error isn't visible when "Use existing customer" is checked so when form submit fails it's not clear why

Reply:
Doesn't need to be, as we set an error on both the customer and email field, so if you have use existing customer checked then you will see the correct "Customer is required" error